fairlearn/reductions/_moments/conditional_selection_rate.py

```diff
@@ -105,12 +116,15 @@ def signed_weights(self, lambda_vec):
         :param lambda_vec: The vector of Lagrange multipliers indexed by `index`
         :type lambda_vec: :class:`pandas:pandas.Series`
         """
-        lambda_signed = lambda_vec["+"] - lambda_vec["-"]
-        adjust = lambda_signed.sum(level=_EVENT) / self.prob_event \
-            - lambda_signed / self.prob_group_event
+        lambda_event = (lambda_vec["+"] - self.ratio * lambda_vec["-"]) \
+            .sum(level=_EVENT) / self.prob_event
+        lambda_group_event = (self.ratio * lambda_vec["+"] - lambda_vec["-"]) / \
+            self.prob_group_event
+        adjust = lambda_event - lambda_group_event
         signed_weights = self.tags.apply(
             lambda row: adjust[row[_EVENT], row[_GROUP_ID]], axis=1
         )
+        signed_weights = self.multiplier.mul(signed_weights)
```

Comment:
This is incorrect for generic `event`--I think it's only correct if `g` is independent of `event`. I think we should rework the derivation, so we can handle arbitrary `g` and `event`.

Reply:
So for existing moments we always have `g` independent of `event`, and that's why this never came up?

Do you think this is in scope for this PR?